packages/next/server/config.ts

```diff
@@ -433,13 +457,13 @@ function assignDefaults(userConfig: { [key: string]: any }) {
     }
 
     if (i18n.domains) {
-      const invalidDomainItems = i18n.domains.filter((item) => {
+      const invalidDomainItems = i18n.domains.filter((item: DomainLocale) => {
```

Comment:
Why are types being added here? The type should be inferred automatically

Reply:
They were being inferred as `any`, even though `copy` returned a `NextConfig`. I think if we updated `defaultConfig` to be a `NextConfigComplete` and copy to also return a `NextConfigComplete` we'd get better inference.